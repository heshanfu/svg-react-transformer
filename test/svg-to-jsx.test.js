'use strict';

const fs = require('fs');
const path = require('path');
const svgToJsx = require('../lib/svg-to-jsx');

const getFixture = name => {
  return fs.readFileSync(
    path.join(__dirname, 'fixtures', `${name}.svg`),
    'utf8'
  );
};

describe('svgToJsx', () => {
  test('works on an airplane', () => {
    return svgToJsx(getFixture('airplane')).then(result => {
      expect(result).toMatchSnapshot();
    });
  });

  test('works on an apple', () => {
    return svgToJsx(getFixture('apple')).then(result => {
      expect(result).toMatchSnapshot();
    });
  });

  test('works on a big one', () => {
    return svgToJsx(getFixture('big')).then(result => {
      expect(result).toMatchSnapshot();
    });
  });

  test('works on one with style attributes', () => {
    return svgToJsx(getFixture('style-attributes')).then(result => {
      expect(result).toMatchSnapshot();
    });
  });

  test('passes SVGO plugins', () => {
    const options = {
      svgoPlugins: [{ removeDimensions: true }]
    };
    return svgToJsx(getFixture('apple'), options).then(result => {
      expect(result).toMatchSnapshot();
    });
  });

  test('catches SVGO error', () => {
    return expect(svgToJsx('#<foo##p')).rejects.toMatch('Error in parsing SVG');
  });
});
