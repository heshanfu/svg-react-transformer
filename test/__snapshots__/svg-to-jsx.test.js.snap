// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`svgToJsx passes SVGO plugins 1`] = `
"<svg
  key=\\"0\\"
  viewBox=\\"0 0 18 18\\"
>
  <path d=\\"M3.4 9.2c.1-1.1.4-2.3 1.5-3.2.9-.7 1.8-.8 2.9-.5 1.4.5 1 .5 2.4 0 1.5-.6 3 0 3.7.8.1.2.2.3 0 .4-1.8 1.3-1.6 3.9.3 5 .2.1.3.2.2.4-.4 1-1 1.9-1.7 2.7-.5.5-1.1.7-1.8.4-.1 0-.3-.1-.4-.1-.9-.1-1.7-.1-2.5.2-.3.1-.5.2-.8.3-.4.1-.8-.1-1.1-.4-.7-.5-1.1-1.2-1.5-1.9-.7-1.1-1.2-2.7-1.2-4.1\\" />
  <path d=\\"M11.6 2.5c0 1.2-1 2.4-2 2.7-.6.1-.8 0-.7-.6.2-1.2 1.2-2.2 2.5-2.5.2 0 .2 0 .2.2v.2\\" />
</svg>"
`;

exports[`svgToJsx works on a big one 1`] = `
"<svg
  key=\\"0\\"
  height=\\"324\\"
  viewBox=\\"0 0 414 324\\"
  width=\\"414\\"
>
  <path
    d=\\"M.333.333h414v324h-414z\\"
    fill=\\"#FFF\\"
  />
  <path
    d=\\"M152.333 24.298h127.432v109.036H152.333z\\"
    fill=\\"none\\"
  />
  <switch>
    <foreignobject
      height=\\"1\\"
      overflow=\\"visible\\"
      requiredExtensions=\\"http://ns.adobe.com/Flows/1.0/\\"
      width=\\"1\\"
      x=\\"0\\"
      y=\\"0\\"
    >
      <flowdef xmlns=\\"http://ns.adobe.com/Flows/1.0/\\">
        <region textmatrix=\\"1 0 0 1 0 0\\">
          <path
            d=\\"M279.765 133.333h-130.49V24.298h130.49v109.035z\\"
            fill=\\"none\\"
          />
        </region>
        <flow xmlns=\\"http://ns.adobe.com/Flows/1.0/\\">
          <p
            text-align=\\"right\\"
            text-align-last=\\"right\\"
          >
            <span
              fontFamily=\\"'TimesNewRomanPS-BoldMT'\\"
              fontSize=\\"9\\"
            >
              ##3$$,
            </span>
            <span
              fontFamily=\\"'TimesNewRomanPSMT'\\"
              fontSize=\\"8.894\\"
            >
              <span
                fontFamily=\\"'TimesNewRomanPSMT'\\"
                fontSize=\\"7\\"
              >
                ##2$$
              </span>
            </span>
          </p>
          <p
            text-align=\\"right\\"
            text-align-last=\\"right\\"
          >
            <span
              fontFamily=\\"'TimesNewRomanPSMT'\\"
              fontSize=\\"8\\"
            >
              ##1$$,
            </span>
            <span
              fontFamily=\\"'TimesNewRomanPSMT'\\"
              fontSize=\\"8.596\\"
            >
              <span
                fontFamily=\\"'TimesNewRomanPSMT'\\"
                fontSize=\\"7\\"
              >
                ##5$$
              </span>
            </span>
          </p>
          <p
            text-align=\\"right\\"
            text-align-last=\\"right\\"
          >
            <span
              fontFamily=\\"'TimesNewRomanPSMT'\\"
              fontSize=\\"8\\"
            >
              ##10$$
            </span>
          </p>
          <p
            text-align=\\"right\\"
            text-align-last=\\"right\\"
          >
            <span
              fontFamily=\\"'TimesNewRomanPSMT'\\"
              fontSize=\\"8\\"
            >
              ##11$$
            </span>
          </p>
          <p
            text-align=\\"right\\"
            text-align-last=\\"right\\"
          >
            <span
              fontFamily=\\"'TimesNewRomanPSMT'\\"
              fontSize=\\"8\\"
            >
              ##7$$
            </span>
          </p>
          <p
            text-align=\\"right\\"
            text-align-last=\\"right\\"
          >
            <span
              fontFamily=\\"'TimesNewRomanPSMT'\\"
              fontSize=\\"8\\"
            >
              ##6$$
            </span>
          </p>
          <p
            text-align=\\"right\\"
            text-align-last=\\"right\\"
          >
            <span
              fontFamily=\\"'TimesNewRomanPSMT'\\"
              fontSize=\\"8\\"
            >
              ##8$$
            </span>
          </p>
          <p
            text-align=\\"right\\"
            text-align-last=\\"right\\"
          >
            <span
              fontFamily=\\"'TimesNewRomanPSMT'\\"
              fontSize=\\"8\\"
            >
              ##4$$
            </span>
          </p>
          <p
            text-align=\\"right\\"
            text-align-last=\\"right\\"
          >
            <span
              fontFamily=\\"'TimesNewRomanPSMT'\\"
              fontSize=\\"8\\"
            >
              ##17$$, ##18$$ ##19$$
            </span>
          </p>
        </flow>
      </flowdef>
    </foreignobject>
    <g>
      <path
        d=\\"M149.275 24.298h130.49v109.036h-130.49z\\"
        fill=\\"none\\"
      />
      <text transform=\\"translate(235.292 30.48)\\">
        <tspan
          fontFamily=\\"'TimesNewRomanPS-BoldMT'\\"
          fontSize=\\"9\\"
          x=\\"0\\"
          y=\\"0\\"
        >
          ##3$$,
        </tspan>
        <tspan
          fontFamily=\\"'TimesNewRomanPSMT'\\"
          fontSize=\\"8.894\\"
          x=\\"24.75\\"
          y=\\"0\\"
        >
           
        </tspan>
        <tspan
          fontFamily=\\"'TimesNewRomanPSMT'\\"
          fontSize=\\"7\\"
          x=\\"26.974\\"
          y=\\"0\\"
        >
          ##2$$
        </tspan>
        <tspan
          fontFamily=\\"'TimesNewRomanPSMT'\\"
          fontSize=\\"8\\"
          x=\\"2.825\\"
          y=\\"10.314\\"
        >
          ##1$$,
        </tspan>
        <tspan
          fontFamily=\\"'TimesNewRomanPSMT'\\"
          fontSize=\\"8.596\\"
          x=\\"24.825\\"
          y=\\"10.314\\"
        >
           
        </tspan>
        <tspan
          fontFamily=\\"'TimesNewRomanPSMT'\\"
          fontSize=\\"7\\"
          x=\\"26.974\\"
          y=\\"10.314\\"
        >
          ##5$$
        </tspan>
        <tspan
          fontFamily=\\"'TimesNewRomanPSMT'\\"
          fontSize=\\"8\\"
          x=\\"20.474\\"
          y=\\"19.915\\"
        >
          ##10$$
        </tspan>
        <tspan
          fontFamily=\\"'TimesNewRomanPSMT'\\"
          fontSize=\\"8\\"
          x=\\"20.474\\"
          y=\\"29.515\\"
        >
          ##11$$
        </tspan>
        <tspan
          fontFamily=\\"'TimesNewRomanPSMT'\\"
          fontSize=\\"8\\"
          x=\\"24.474\\"
          y=\\"39.115\\"
        >
          ##7$$
        </tspan>
        <tspan
          fontFamily=\\"'TimesNewRomanPSMT'\\"
          fontSize=\\"8\\"
          x=\\"24.474\\"
          y=\\"48.715\\"
        >
          ##6$$
        </tspan>
        <tspan
          fontFamily=\\"'TimesNewRomanPSMT'\\"
          fontSize=\\"8\\"
          x=\\"24.474\\"
          y=\\"58.314\\"
        >
          ##8$$
        </tspan>
        <tspan
          fontFamily=\\"'TimesNewRomanPSMT'\\"
          fontSize=\\"8\\"
          x=\\"24.474\\"
          y=\\"67.915\\"
        >
          ##4$$
        </tspan>
        <tspan
          fontFamily=\\"'TimesNewRomanPSMT'\\"
          fontSize=\\"8\\"
          x=\\"-33.526\\"
          y=\\"77.515\\"
        >
          ##17$$, ##18$$ ##19$$
        </tspan>
      </text>
    </g>
  </switch>
  <path
    d=\\"M288.016 17.766h109.256v87.97H288.016z\\"
    fill=\\"#251413\\"
  />
  <path
    d=\\"M46.389 234.756h13.958v13.958H46.389zM46.389 218.081h13.958v13.958H46.389z\\"
    fill=\\"#4C2525\\"
  />
  <g>
    <path
      d=\\"M66.715 220.205h115.852v30.135H66.715z\\"
      fill=\\"none\\"
    />
    <switch>
      <foreignobject
        height=\\"1\\"
        overflow=\\"visible\\"
        requiredExtensions=\\"http://ns.adobe.com/Flows/1.0/\\"
        width=\\"1\\"
        x=\\"0\\"
        y=\\"0\\"
      >
        <flowdef xmlns=\\"http://ns.adobe.com/Flows/1.0/\\">
          <region textmatrix=\\"1 0 0 1 0 0\\">
            <path
              d=\\"M190.394 245.59H66.715v-25.385h123.679v25.385z\\"
              fill=\\"none\\"
            />
          </region>
          <flow
            fontFamily=\\"'TimesNewRomanPSMT'\\"
            fontSize=\\"5\\"
            xmlns=\\"http://ns.adobe.com/Flows/1.0/\\"
          >
            <p>
              <span>
                ##25$$
              </span>
            </p>
          </flow>
        </flowdef>
      </foreignobject>
      <g>
        <path
          d=\\"M66.715 220.205h123.679v25.385H66.715z\\"
          fill=\\"none\\"
        />
        <text
          fontFamily=\\"'TimesNewRomanPSMT'\\"
          fontSize=\\"5\\"
          transform=\\"translate(66.715 223.68)\\"
        >
          ##25$$
        </text>
      </g>
    </switch>
  </g>
</svg>"
`;

exports[`svgToJsx works on an airplane 1`] = `
"<svg
  key=\\"0\\"
  viewBox=\\"0 0 18 18\\"
>
  <path d=\\"M7 4l1.6 4H5.5S4.1 6 3 6h-.8L3 8l1 3h4.6L7 15h2l3.2-4H14c1 0 2-.7 2-1.5S15 8 14 8h-1.8L9 4H7z\\" />
</svg>"
`;

exports[`svgToJsx works on an apple 1`] = `
"<svg
  key=\\"0\\"
  height=\\"30\\"
  viewBox=\\"0 0 18 18\\"
  width=\\"30\\"
>
  <path d=\\"M3.4 9.2c.1-1.1.4-2.3 1.5-3.2.9-.7 1.8-.8 2.9-.5 1.4.5 1 .5 2.4 0 1.5-.6 3 0 3.7.8.1.2.2.3 0 .4-1.8 1.3-1.6 3.9.3 5 .2.1.3.2.2.4-.4 1-1 1.9-1.7 2.7-.5.5-1.1.7-1.8.4-.1 0-.3-.1-.4-.1-.9-.1-1.7-.1-2.5.2-.3.1-.5.2-.8.3-.4.1-.8-.1-1.1-.4-.7-.5-1.1-1.2-1.5-1.9-.7-1.1-1.2-2.7-1.2-4.1\\" />
  <path d=\\"M11.6 2.5c0 1.2-1 2.4-2 2.7-.6.1-.8 0-.7-.6.2-1.2 1.2-2.2 2.5-2.5.2 0 .2 0 .2.2v.2\\" />
</svg>"
`;

exports[`svgToJsx works on one with style attributes 1`] = `
"<svg
  key=\\"0\\"
  height=\\"69.855\\"
  viewBox=\\"0 0 79.4 69.855\\"
  width=\\"79.4\\"
>
  <defs>
    <clippath
      id=\\"a\\"
      transform=\\"translate(-2.3 -8.8)\\"
    >
      <rect
        fill=\\"none\\"
        height=\\"43\\"
        rx=\\"4.5\\"
        ry=\\"4.5\\"
        width=\\"60\\"
        x=\\"2.3\\"
        y=\\"24\\"
      />
    </clippath>
  </defs>
  <title>
    data-viz-maps
  </title>
  <path
    d=\\"M54.9 28.2l8.3-1.5a1.817 1.817 0 0 1 2.1 1.4l5.3 29a1.817 1.817 0 0 1-1.4 2.1l-8.3 1.5a1.817 1.817 0 0 1-2.1-1.4l-5.3-29a1.743 1.743 0 0 1 1.4-2.1z\\"
    fill=\\"#c5b8ed\\"
  />
  <path
    d=\\"M79.4 36.4L77.6 29a1.8 1.8 0 0 0-2.1-1.3c-.2 0-.3.1-.5.2l-6.7 3.8-9.1 20.9a3.189 3.189 0 0 0 1.6 4.2l5.1 2.3a3.189 3.189 0 0 0 4.2-1.6l8.7-20z\\"
    fill=\\"#dad2f9\\"
  />
  <path
    d=\\"M60.857 37.259l5.901-1.085.362 1.967-5.901 1.085zM59.96 32.358l5.901-1.085.362 1.967-5.901 1.085z\\"
    fill=\\"#dad2f9\\"
  />
  <circle
    cx=\\"66.896\\"
    cy=\\"61.496\\"
    fill=\\"#dad2f9\\"
    r=\\"2.5\\"
    transform=\\"rotate(-10.422 17.502 69.707)\\"
  />
  <path
    d=\\"M58.2 48.2h13.6a3.159 3.159 0 0 1 3.2 3.2V65a3.159 3.159 0 0 1-3.2 3.2H58.2A3.159 3.159 0 0 1 55 65V51.4a3.222 3.222 0 0 1 3.2-3.2z\\"
    fill=\\"#c5b8ed\\"
  />
  <path
    d=\\"M21 58.2h19v4H21z\\"
    fill=\\"#5a39c3\\"
  />
  <path
    d=\\"M4.5 15.2h51a4.481 4.481 0 0 1 4.5 4.5v34a4.481 4.481 0 0 1-4.5 4.5h-51A4.481 4.481 0 0 1 0 53.7v-34a4.481 4.481 0 0 1 4.5-4.5z\\"
    fill=\\"#5a39c3\\"
  />
  <path
    d=\\"M22.1 45.9a2.342 2.342 0 0 1-1.7.7h-5a2.06 2.06 0 0 0-1.7.7l-3 2.9h49.4V20.1a4.7 4.7 0 0 0-4.5-4.9H23.5v9.9a2.456 2.456 0 0 1-2.5 2.4h-3.5a2.523 2.523 0 0 0-2.5 2.4V34a2.609 2.609 0 0 0 .7 1.7l6.6 6.5a2.357 2.357 0 0 1 .1 3.3l-.1.1z\\"
    fill=\\"#774cef\\"
    opacity=\\".5\\"
    style={{
      isolation: 'isolate'
    }}
  />
  <g clipPath=\\"url(#a)\\">
    <path
      d=\\"M53 10.2l-1.2 1.6\\"
      fill=\\"none\\"
      stroke=\\"#774cef\\"
      strokeMiterlimit=\\"10\\"
      strokeWidth=\\"2\\"
    />
  </g>
  <g clipPath=\\"url(#a)\\">
    <path
      d=\\"M24.3 67.7L23 69.2\\"
      fill=\\"none\\"
      stroke=\\"#774cef\\"
      strokeMiterlimit=\\"10\\"
      strokeWidth=\\"2\\"
    />
  </g>
  <g clipPath=\\"url(#a)\\">
    <path
      d=\\"M66 13.2l-1.2 1.6\\"
      fill=\\"none\\"
      stroke=\\"#774cef\\"
      strokeMiterlimit=\\"10\\"
      strokeWidth=\\"2\\"
    />
  </g>
  <path
    d=\\"M0 50.2h60v3.5a4.481 4.481 0 0 1-4.5 4.5h-51A4.481 4.481 0 0 1 0 53.7v-3.5z\\"
    fill=\\"#774cef\\"
  />
  <circle
    cx=\\"30.5\\"
    cy=\\"54.7\\"
    fill=\\"#5a39c3\\"
    r=\\"1.5\\"
  />
  <path
    d=\\"M18 62.2h26v6H18z\\"
    fill=\\"#774cef\\"
  />
  <circle
    cx=\\"21.6\\"
    cy=\\"29.1\\"
    fill=\\"#c5b8ed\\"
    opacity=\\".5\\"
    r=\\"5.8\\"
    style={{
      isolation: 'isolate'
    }}
  />
  <circle
    cx=\\"31.9\\"
    cy=\\"40.5\\"
    fill=\\"#c5b8ed\\"
    opacity=\\".5\\"
    r=\\"1.4\\"
    style={{
      isolation: 'isolate'
    }}
  />
  <circle
    cx=\\"47.8\\"
    cy=\\"31.6\\"
    fill=\\"#c5b8ed\\"
    opacity=\\".5\\"
    r=\\"1.4\\"
    style={{
      isolation: 'isolate'
    }}
  />
  <circle
    cx=\\"31\\"
    cy=\\"24.5\\"
    fill=\\"#c5b8ed\\"
    opacity=\\".5\\"
    r=\\"1.4\\"
    style={{
      isolation: 'isolate'
    }}
  />
  <circle
    cx=\\"21.6\\"
    cy=\\"29.1\\"
    fill=\\"#dad2f9\\"
    opacity=\\".5\\"
    r=\\"3.2\\"
    style={{
      isolation: 'isolate'
    }}
  />
  <circle
    cx=\\"40.5\\"
    cy=\\"42.7\\"
    fill=\\"#c5b8ed\\"
    opacity=\\".5\\"
    r=\\"3.5\\"
    style={{
      isolation: 'isolate'
    }}
  />
  <circle
    cx=\\"40.5\\"
    cy=\\"42.7\\"
    fill=\\"#dad2f9\\"
    opacity=\\".5\\"
    r=\\"1.7\\"
    style={{
      isolation: 'isolate'
    }}
  />
  <path
    d=\\"M70.2 13l-1.6 3.2-3.2 1.6 3.2 1.5 1.6 3.3 1.6-3.3 3.2-1.5-3.2-1.6-1.6-3.2zM49.2 0l-1.6 3.2-3.2 1.6 3.2 1.5 1.6 3.3 1.6-3.3L54 4.8l-3.2-1.6L49.2 0z\\"
    fill=\\"#dad2f9\\"
  />
  <path
    d=\\"M79.4 36.4a4.822 4.822 0 0 1-3.565-2.513 4.342 4.342 0 0 0-5.695-2.427 5.918 5.918 0 0 1-1.87.32l.03-.08 6.7-3.8c.2-.1.3-.2.5-.2a1.8 1.8 0 0 1 2.1 1.3z\\"
    fill=\\"#c5b8ed\\"
  />
</svg>"
`;
